source/appModules/explorer.py

```diff
@@ -177,6 +179,14 @@ def _get_windowText(self):
 			return windowText.replace(CHAR_LTR_MARK,'').replace(CHAR_RTL_MARK,'')
 		return windowText
 
+
+class metadataEditField(RichEdit50):
+	""" Used for metadata edit fields in Windows Explorer in Windows 7.
+	By default these fields would use ITextDocumentTextInfo ,
+	but to avoid Windows Explorer crashes we need to use EditTextInfo here. """
+	def _get_TextInfo(self):

```

Comment:
I think this is a perfect case to benefit from magic class properties:

```suggestion
	@classmethod
	def _get_TextInfo(cls):
```

See also the suggestion for the code below. This ensures that:

1. The winVersion check is performed only once, rather than multiple times
2. The appropriate TextInfo class is cached for the lifetime of the nVDA instance.